spec/controllers/summary_controller_spec.rb

```diff
@@ -362,4 +362,89 @@
       end
     end
   end
+
+  describe '#search' do
+    let(:nomis_staff_id) { 485_926 }
+
+    let(:poms) {
+      [
+        build(:pom,
+              firstName: 'Alice',
+              lastName: 'Ward',
+              position: RecommendationService::PRISON_POM,
+              staffId: nomis_staff_id
+        )
+      ]
+    }
+
+    before do
+      stub_poms(prison, poms)
+      stub_signed_in_pom(prison, nomis_staff_id)
+    end
+
+    context 'when user is an SPO ' do
+      before do
+        stub_sso_data(prison)
+      end
+
+      it 'can search' do
+        offenders = build_list(:nomis_offender, 1)
+        stub_offenders_for_prison(prison, offenders)
+
+        get :search, params: { prison_id: prison, q: 'Cal' }
+        expect(response.status).to eq(200)
+        expect(response).to be_successful
+
+        expect(assigns(:q)).to eq('Cal')
+        expect(assigns(:offenders).size).to eq(0)
+      end
+
+      context "with 3 offenders", :allocation do
+        let(:offenders) { build_list(:nomis_offender, 3, lastName: 'Bloggs', firstName: 'Alice') }
+        let(:updated_offenders) { assigns(:offenders) }
+        let(:alloc_offender) { updated_offenders.detect { |o| o.offender_no == offenders.first.fetch(:offenderNo) } }
```

Comment:
Stephen, I am trying to understand what alloc_offender does. It compares the first offender from offenders list with the updated_offenders and returns the updated_offenders match. This is then assigns to the name alloc_offender. Oh I think I am confused!! How does this make it allocated?

Reply:
It's not great I agree It's trying to pick out the first one as an 'offender to be allocated' I think. Let me try and re-work it